### PR TITLE
Feature 13 survey status

### LINKED ISSUE #13: Survey Status
As a user I'd like that Surveys would have a field, status, to define whether they're in approved or draft state.

### LINKED ISSUE #14: Abstract database operations awaiting

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #13 
Resolves #14

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/oss/api/impl/surveys/SurveyController.kt [hunk 1]
```diff
@@ -19,9 +20,22 @@ class SurveyController {
...
      * @return uni with list of surveys and count
      */
-    suspend fun listSurveys(firstResult: Int?, maxResults: Int?): Pair<List<SurveyEntity>, Long> {
+    suspend fun listSurveys(
+        firstResult: Int?,
+        maxResults: Int?,
+        status: SurveyStatus?
+    ): Pair<List<SurveyEntity>, Long> {
+        if (status != null) {
+            return surveyRepository.applyPagingToQuery(
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/oss/api/impl/surveys/SurveyController.kt:32]: Shouldn't applyPagingToQuery be used internally by the repository?

File: src/main/resources/db/changeLog.xml [hunk 2]
```diff
@@ -73,4 +73,12 @@
             </column>
         </createTable>
     </changeSet>
+
+    <changeSet id="survey_status" author="Ville Juutila">
```
Comment (kaeline) [inline src/main/resources/db/changeLog.xml:77]: Since it's not nullable property, should we add a query to assign some status to existing surveys?

**Review by kaeline — APPROVED** [review]

**Review by anttileppa — APPROVED** [review]

File: src/main/resources/db/changeLog.xml (hunk 2 above)
Comment (villejuutila) [reply src/main/resources/db/changeLog.xml:77]: Resolved.

File: src/main/kotlin/fi/metatavu/oss/api/impl/surveys/SurveyController.kt (hunk 1 above)
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/oss/api/impl/surveys/SurveyController.kt]: Resolved.

**Pull request merged**